source/inject.ts

```diff
@@ -22,88 +23,122 @@ const injectAndDiscardCertainErrors: typeof injectContentScript = async (tabId,
 	}
 };
 
-function forgetTab(tabId: number) {
-	tracked.delete(tabId);
-	if (tracked.size === 0) {
-		chrome.tabs.onUpdated.removeListener(onDiscarded);
-		chrome.tabs.onRemoved.removeListener(forgetTab);
-		chrome.tabs.onActivated.removeListener(onActivated);
-		chrome.webNavigation?.onCommitted.removeListener(onCommitted);
-	}
+async function forgetTab(tabId: number) {
+	await removeTabFromWaitingList(tabId);
+	await removeListenersIfDone();
```

Comment:
It doesn't look like `requestIdleCallback` is available in the service worker

<img width="517" alt="Screenshot 4" src="https://github.com/user-attachments/assets/94fb731c-2dc0-4db6-8916-3dd1909eebf5" />